htlcswitch/interfaces.go

```diff
@@ -117,12 +117,15 @@ type ChannelLink interface {
 	CheckHtlcTransit(payHash [32]byte, amt lnwire.MilliSatoshi,
 		timeout uint32, heightNow uint32) lnwire.FailureMessage
 
-	// Bandwidth returns the amount of milli-satoshis which current link
-	// might pass through channel link. The value returned from this method
-	// represents the up to date available flow through the channel. This
-	// takes into account any forwarded but un-cleared HTLC's, and any
-	// HTLC's which have been set to the over flow queue.
-	Bandwidth() lnwire.MilliSatoshi
+	// Bandwidth returns the total amount that can flow through the channel
+	// link at this given instance. The amt parameter should be set
+	// non-zero to evaluate whether the link can carry this amt as part of
+	// its bandwidth. The value returned is expressed in millisatoshi and
+	// can be used by callers when making forwarding decisions to determine
+	// if a link can accept an HTLC of this size. This takes into account
+	// any forwarded but un-cleared HTLCs, and any HTLCs which have been
+	// set to the over flow queue.
+	Bandwidth(lnwire.MilliSatoshi) lnwire.MilliSatoshi
```

Comment:
Does it need this parameter, or should the calculation just return the highest amt that we can send without getting into trouble?

Reply:
That sounds like a much better way of solving it indeed. Will rework the PR to try that.